### PR TITLE
Load the prelude and additional modules when the repl starts

### PR DESCRIPTION
**Pull request by boomshroom**
This is a step towards #17, but further work is blocked on compiler internals.

Simply put, the prelude and any other files passed by the command line are loaded into the REPL's environment. Currently, files loaded later will include previous files' namespaces. In addition, only values with a predeclaration are added to the environment. Further investigation will be needed, though this behavior will change when exports are properly added.

### CODE REVIEW

File: src/cli/repl.rs [hunk 1]
```diff
@@ -235,6 +260,30 @@ fn eval_print(
     Ok(ControlFlow::Continue)
 }
 
+
+fn load_module(m: Module, tc_env: &mut TcEnv, codemap: &CodeMap, writer: &StandardStream) {
```
Comment (brendanzab) [inline src/cli/repl.rs]: What about instead trying to make the API in `lib.rs` more flexible? I'm thinking something like:

```rust
pub fn load_file(tc_env: &mut TcEnv, tc_Env file: &FileMap) -> Result<core::Module, Vec<Diagnostic>> {
```

Would that work?

Comment (brendanzab) [reply src/cli/repl.rs]: Yup. Well, you could make a copy of the `TcEnv` I guess?

Comment (brendanzab) [reply src/cli/repl.rs]: Still haven't thought through how this compiler pipeline would look like, thanks for forcing me to think about it! 😅

**Comment by brendanzab** [comment]
Sorry for taking my time on this one, still thinking about it!

**Comment by brendanzab** [comment]
Ok, so I've removed top level modules in #149 - wondering if instead we could just load the prelude record into a binding called `prelude`. Eventually we could think of ways to make that nicer, but it might be a good start! At least you'd then be able to call `prelude.id` in the repl!

**Comment by boomshroom** [comment]
Usually `prelude` is for values that shouldn't require qualifying. In rust, you can just use `Ok` and `Err` without having to qualify them as `Result::Ok` or `Result::Err` because they're so common.

**Comment by brendanzab** [comment]
Yeah, definitely! That's what I meant by 'making it easier'

**Comment by brendanzab** [comment]
I'm wondering if it would be nice to have repl init scripts - like, you could have something like:

```
:let record { .. } = import prelude
```

**Pull request closed without merging**